rpcserver.go

```diff
@@ -1103,3 +1108,75 @@ func (r *rpcServer) SendAsset(ctx context.Context,
 		TotalFeeSats: int64(resp.TotalFees),
 	}, nil
 }
+
+// SubscribeSendAssetEventNtfns registers a subscription to the event
+// notification stream which relates to the asset sending process.
+func (r *rpcServer) SubscribeSendAssetEventNtfns(
+	in *tarorpc.SubscribeSendAssetEventNtfnsRequest,
+	ntfnStream tarorpc.Taro_SubscribeSendAssetEventNtfnsServer) error {
+
+	// Create a new event subscriber and pass a copy to the chain porter.
+	// We will then read events from the subscriber.
+	eventSubscriber := chanutils.NewEventReceiver[tarofreighter.Event](
+		chanutils.DefaultQueueSize,
+	)
+	defer eventSubscriber.Stop()
+
+	err := r.cfg.ChainPorter.RegisterSubscriber(eventSubscriber, false, false)
+	if err != nil {
+		return fmt.Errorf("failed to register event notifications "+
+			"subscription: %w", err)
+	}
+
+	// Loop and read from the ChainPorter event subscription and forward to
+	// the RPC stream.
+	for {
+		select {
+		// Handle receiving a new event from the ChainPorter.
+		// The event will be mapped to the RPC event type and
+		// sent over the stream.
+		case event := <-eventSubscriber.NewItemCreated.ChanOut():
+
+			rpcEvent, err := marshallSendAssetEvent(event)
+			if err != nil {
+				return fmt.Errorf("failed to marshall "+
+					"ChainPorter event into RPC event: "+
+					"%w", err)
+			}
+
+			err = ntfnStream.Send(rpcEvent)
+			if err != nil {
+				return fmt.Errorf("failed to RPC stream send "+
+					"event: %w", err)
+			}
+
+		// Handle the case where the RPC stream is closed by the
+		// client.
+		case <-ntfnStream.Context().Done():
+			return nil
+
+		// Handle the case where the RPC server is shutting down.
+		case <-r.quit:
+			return nil
+		}
+	}
+}
+
+// marshallSendAssetEvent maps a ChainPorter event to its RPC counterpart.
+func marshallSendAssetEvent(eventInterface tarofreighter.Event) (*tarorpc.SendAssetEvent, error) {
```

Comment:
nit: line length.

Reply:
Should I format as follows:
```
func marshallSendAssetEvent(eventInterface tarofreighter.Event) (
	*tarorpc.SendAssetEvent, error) {

	switch event := eventInterface.(type) {
```	
I'm trying to avoid the dangling `(`.

Reply:
In this case I'd probably break before the argument and then again before the `error` (if necessary).

Reply:
Good to know, thanks.